### PR TITLE
Feature multiarray support doc

### PR DESCRIPTION
**Pull request by medegw01**
# Description
This PR covers the following:

-  document multiarray support and installation
-  fully test sidecar injection for powermax (Valid volume creation, quota checks,)
-  modifies sidecar inject to support powermax reverse-proxy(authorization now works with or without reverse-proxy)
- Add necessary unit tests

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
|     #92      |

# Checklist:

- [x] I have performed a self-review of my own changes.

### CODE REVIEW

**Comment by ioeicicdsvc** [comment]
@medegw01: Thank you for submitting your pull request!

Please make sure you have followed our contributing guidelines. We will review it as soon as possible.

------------------------------------------------------------------------------------------------------------------------
We want your feedback! If you have any questions or suggestions regarding our contributing process/workflow, please reach out to us at karavi@dell.com.

File: cmd/karavictl/cmd/inject.go [hunk 1]
```diff
@@ -579,6 +585,112 @@ func (lc *ListChangeForPowerMax) injectKaraviSecret(insecure bool) {
...
+		lc.Err = err
+		return
+	}
+
+	deploy, ok := m[lc.InjectResources.ReverseProxy]
+	if !ok {
+		return
+	}
+
+	// Set configMAP
```
Comment (hoppea2) [inline cmd/karavictl/cmd/inject.go:608]: buildMapOfCofigMap?

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_new.yaml [hunk 2]
```diff
@@ -487,7 +672,7 @@ items:
             name: socket-dir
         - args:
           - --csi-address=$(ADDRESS)
-          - --volume-name-prefix=pmax
+          - --volume-name-prefix=mich
```
Comment (hoppea2) [inline cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_new.yaml:675]: keep prefix it generic (pmax)

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml [hunk 3]
```diff
@@ -773,15 +858,15 @@ items:
           - name: X_CSI_POWERMAX_INSECURE
             value: none
           - name: X_CSI_POWERMAX_PROXY_SERVICE_NAME
-            value: none
+            value: powermax-reverseproxy
           - name: X_CSI_POWERMAX_PORTGROUPS
-            value: powermax-port-group
+            value: iscsi_lglw3021
```
Comment (hoppea2) [inline cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml:863]: revert to powermax-port-group

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml [hunk 4]
```diff
@@ -808,7 +893,7 @@ items:
             readOnly: true
         - env:
           - name: PROXY_HOST
-            value: proxy-host
+            value: lglbv161.lss.emc.com
```
Comment (hoppea2) [inline cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml:896]: remove host

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml [hunk 5]
```diff
@@ -823,7 +908,7 @@ items:
               secretKeyRef:
                 key: refresh
                 name: proxy-authz-tokens
-          image: sidecar-proxy:latest
+          image: amaas-eos-mw1.cec.lab.emc.com:5046/sidecar-proxy:latest
           imagePullPolicy: Always
```
Comment (hoppea2) [inline cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml:912]: remove artifactory

File: docs/GETTING_STARTED_GUIDE.md [hunk 6]
```diff
@@ -303,69 +304,101 @@ This will also provide code coverage statistics for the various Go packages.
 
 ## Configuring CSI Driver with Authorization
 
-Given a setup where Kubernetes, a storage system, a CSI driver, and Karavi Authorization are deployed, follow the steps below to configure the the CSI Driver with Authorization:
+Given a setup where Kubernetes, a storage system, CSI driver(s), and Karavi Authorization are deployed, follow the steps below to configure the CSI Drivers to work  Authorization sidecar:
```
Comment (hoppea2) [inline docs/GETTING_STARTED_GUIDE.md:307]: *with

File: docs/GETTING_STARTED_GUIDE.md [hunk 7]
```diff
@@ -377,4 +410,4 @@ To test the setup, follow the steps below:
 - Create a PVC request from the StorageClass with any storage capacity less than the RoleQuota you specified during configuration
 - Request a Pod to consume the PVC created above. If everything is well configured, the PVC will be bound to storage and the volume will be created on on the storage system.
 
-You can also test failure case, buy repeating the above steps but specify a quota larger than RoleQuota you specified. However, when you request a Pod to use PVC, you'll get request is denied as PVC exceeds capacity and pv will be in a pending state.
+You can also test failure case, buy repeating the above steps but specify a quota larger than RoleQuota you specified. Conversely, when you request a Pod to use PVC, you'll get request is denied as PVC exceeds capacity and pv will be in a pending state.
```
Comment (hoppea2) [inline docs/GETTING_STARTED_GUIDE.md:413]: typo: buy
capitalize: PV

File: internal/proxy/powerflex_handler.go [hunk 8]
```diff
@@ -319,6 +319,7 @@ func (s *System) volumeCreateHandler(next http.Handler, enf *quota.RedisEnforcem
 		if err != nil {
 			s.log.Errorf("proxy: decoding create volume request: %+v", err)
 			writeError(w, "failed to extract cap data", http.StatusBadRequest)
+			time.Sleep(10 * time.Second)
```
Comment (hoppea2) [inline internal/proxy/powerflex_handler.go:322]: remove sleep

File: cmd/karavictl/cmd/inject.go [hunk 9]
```diff
@@ -967,6 +1079,28 @@ func (lc *ListChangeForMultiArray) injectIntoDaemonset(imageAddr, proxyHost stri
 	lc.Modified.Items = append(lc.Modified.Items, raw)
 }
 
+func buildbuildMapOfCofigMap(list *corev1.List) (map[string]*corev1.ConfigMap, error) {
```
Comment (tdawe) [inline cmd/karavictl/cmd/inject.go:1082]: Typo: buildbuildMapOfCofigMap -> buildMapOfConfigMap

Comment (tdawe) [reply cmd/karavictl/cmd/inject.go:1082]: Fixed

File: internal/proxy/powerflex_handler.go (hunk 8 above)
Comment (tdawe) [reply internal/proxy/powerflex_handler.go:322]: Fixed

File: docs/GETTING_STARTED_GUIDE.md (hunk 7 above)
Comment (tdawe) [reply docs/GETTING_STARTED_GUIDE.md:413]: Fixed

File: docs/GETTING_STARTED_GUIDE.md (hunk 6 above)
Comment (tdawe) [reply docs/GETTING_STARTED_GUIDE.md:307]: Fixed

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml (hunk 5 above)
Comment (tdawe) [reply cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml:912]: Fixed

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml (hunk 4 above)
Comment (tdawe) [reply cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml:896]: Fixed

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml (hunk 3 above)
Comment (tdawe) [reply cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_update.yaml:863]: Fixed

File: cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_new.yaml (hunk 2 above)
Comment (tdawe) [reply cmd/karavictl/cmd/testdata/kubectl_get_all_in_powermax_new.yaml:675]: Fixed

File: cmd/karavictl/cmd/inject.go (hunk 1 above)
Comment (tdawe) [reply cmd/karavictl/cmd/inject.go:608]: Fixed

**Review by hoppea2 — APPROVED** [review]

**Review by tdawe — APPROVED** [review]

**Pull request merged**